src/main/java/cz/it4i/parallel/HeappeParadigm.java

```diff
@@ -0,0 +1,116 @@
+package cz.it4i.parallel;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+
+import org.scijava.parallel.ParallelizationParadigm;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import cz.it4i.fiji.haas_java_client.HaaSClient;
+import cz.it4i.fiji.haas_java_client.JobSettingsBuilder;
+import cz.it4i.fiji.haas_java_client.JobState;
+import cz.it4i.fiji.haas_java_client.SettingsProvider;
+import cz.it4i.fiji.haas_java_client.TunnelToNode;
+
+@Plugin(type = ParallelizationParadigm.class)
+public class HeappeParadigm extends SimpleOstravaParadigm {
+
+	private static final int TIMEOUT_BETWEEN_JOB_STATE_QUERY = 1000;
+
+	@Parameter
+	private int port;
+
+	public static final Logger log = LoggerFactory.getLogger(cz.it4i.parallel.HeappeParadigm.class);
+
+	private int numberOfHosts;
+
+	private HaaSClient haasClient;
+	private final Collection<TunnelToNode> tunnels = Collections.synchronizedList(new LinkedList<>());
+
+	// -- HeappeParadigm methods --
+
+	public void setPort(int port) {
+		this.port = port;
+	}
+
+	public void setNumberOfNodes(int number) {
+		this.numberOfHosts = number;
+	}
+
+	// -- SimpleOstravaParadigm methods --
+
+	// -- Closeable methods --
+	
+	@Override
+	public void close() {
+		tunnels.forEach(t -> {
+			try {
+				t.close();
+			} catch (IOException e) {
+				log.error(e.getMessage(), e);
+			}
+		});
+	}
+
+	@Override
+	protected void initWorkerPool() {
+		if (log.isDebugEnabled()) {
+			log.debug("initWorkerPool");
+		}
+		haasClient = new HaaSClient(SettingsProvider.getSettings(Constants.PROJECT_ID, Constants.CONFIG_FILE_NAME));
+		if (log.isDebugEnabled()) {
+			log.debug("createJob");
+		}
+		long jobId = haasClient.createJob(
```

Comment:
Consider making this local variable final. The same suggestions applies to multiple occurrences in this method.

Reply:
According to the discussion https://stackoverflow.com/questions/316352/why-would-one-mark-local-variables-and-method-parameters-as-final-in-java we can discuss it and use it as a  global rule. However, I don't think as an issue of this merge request.
Let's discuss code convention rules and create rules for automatic changes in Eclipse IDE.

Reply:
Agree, created an issue (https://github.com/PetrBainar/scijava-parallel/issues/9) lest we forget it.